src/digest.c

```diff
@@ -72,16 +72,78 @@ FILE* open_with_widechar_on_windows(const char* txt) {
     return out;
 }
 
+
+// Also already used in sha2.h
+//
+// We can rely on WORDS_BIGENDIAN only be defined on big endian systems thanks to Rconfig.
+//
+// A number of other #define based tests are in other source files here for different hash
+// algorithm implementations notably crc32c, pmurhash, sha2 and xxhash
+//
+// A small and elegant test is also in package qs based on https://stackoverflow.com/a/1001373
+
+// edd 02 Dec 2013  use Rconfig.h to define BYTE_ORDER, unless already defined
+#ifndef BYTE_ORDER
+// see sha2.c comments, and on the internet at large
+#define LITTLE_ENDIAN 1234
+#define BIG_ENDIAN    4321
+#ifdef WORDS_BIGENDIAN
+#define BYTE_ORDER  BIG_ENDIAN
+#else
+#define BYTE_ORDER  LITTLE_ENDIAN
+#endif
+#endif
+
+SEXP is_big_endian(void) {
+    return Rf_ScalarLogical(BYTE_ORDER == BIG_ENDIAN);
+}
+
+SEXP is_little_endian(void) {
+    return Rf_ScalarLogical(BYTE_ORDER == LITTLE_ENDIAN);
+}
+
+/* for any algorithm that has output as unsigned char*, can use resolve */
+#define resolve(WHAT) if (leaveRaw) {                          \
```

Comment:
How about renaming to either `_resolve` (to signal internal function) or maybe state more clearly what it is for, ie something like `_cast_from_unsigned_char` or alike?

Reply:
propose: `_store_from_char_ptr`

Reply:
That works.  'store' to me is like committing to disk, so 'return' or 'convert' would work too but all good!